### PR TITLE
Merging NSFRB pipeline updates to main

### PR DESCRIPTION
**Pull request by nkosogor**
@caseyjlaw @mbsherma Could you please review this PR?

@cpviolator I think this might be of interest to you as well. Any feedback or comments you may have would be greatly appreciated!

## Changes and Features Added

- **Tests:**
  - Added unit tests for `imaging.py`, `classifying.py`, `network.py`, and `searching.py`.

- **Documentation:**
  - Updated `README.md` and other documentation.
  - Improved module descriptions.

- **New Modules and Functions:**
  - Added `TXclient.py` for HTTP data transfers.
  - Introduced `classifying.py` with a new CNN for image classification.

- **New Data and Configuration:**
  - Added new candidate data files.
  - Updated configuration settings and introduced `casa38nsfrb_env.yml` for environment setup.

### CODE REVIEW

File: scripts/run_NSFRB
Comment (mbsherma) [inline scripts/run_NSFRB:1]: @mbsherma this file can be deleted; leftover from C implementation

File: scripts/run_classifier.py
Comment (mbsherma) [inline scripts/run_classifier.py:1]: @mbsherma we should delete this file, it still pulls from stdin instead of the process server

File: scripts/kill_NSFRB
Comment (mbsherma) [inline scripts/kill_NSFRB:1]: @mbsherma delete this, replaced with process_server/kill_proc_server

File: scripts/process_log.txt
Comment (mbsherma) [inline scripts/process_log.txt:1]: @mbsherma delete this file, replaced by process_server/process_log.txt

File: process_server/process_log.txt
Comment (mbsherma) [inline process_server/process_log.txt:1]: @mbsherma move log files local/diff repo/dir

File: process_server/run_COMBINED.py
Comment (mbsherma) [inline process_server/run_COMBINED.py]: @mbsherma need to correct how I used concurrent.futures so it doesn't hang after starting each thread

File: process_log.txt
Comment (mbsherma) [inline process_log.txt:1]: @mbsherma remove this, replaced by process_server/process_log.txt

File: tmpoutput/all_cluster_cands.csv
Comment (mbsherma) [inline tmpoutput/all_cluster_cands.csv:1]: @mbsherma @nkosogor  tmpoutput directory is alright for now, but eventually need different plan for saving output files; e.g. should we make a function to update the output directory after installation?

File: tmpoutput/search_log_thread_row2_col1.txt
Comment (mbsherma) [inline tmpoutput/search_log_thread_row2_col1.txt:1]: @mbsherma Need to find a different way to distinguish different threads in log files; separate file for each thread is unsustainable

File: nsfrb/searching.py
Comment (mbsherma) [inline nsfrb/searching.py]: @mbsherma Need to remove dead code here

File: nsfrb/searching.py
Comment (mbsherma) [inline nsfrb/searching.py]: @mbsherma Need to remove dead code here

**Review by mbsherma — COMMENTED** [review]
@caseyjlaw @nkosogor @cpviolator Overall there's no merge conflicts after my first review; however, I've found a few instances of dead code that I need to remove, and multiple log files I've made that are cluttering the repo. I've commented these above and assigned them to myself to resolve. @nkosogor I'm going to work on removing some dead code and clearing the log files and can approve once i've done that. Once its merged I'll take a closer look at how I'm maintaining the log files. There's also a small bug in the process server with how I'm using concurrent.futures that I'll fix before we merge

File: nsfrb/searching.py
Comment (mbsherma) [reply nsfrb/searching.py]: Resolved in commit [dc6ae47](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/dc6ae47901cac08f0963bcbbfc0928c969a64c15)

File: nsfrb/searching.py
Comment (mbsherma) [reply nsfrb/searching.py]: Resolved in commit [dc6ae47](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/dc6ae47901cac08f0963bcbbfc0928c969a64c15)

File: process_log.txt
Comment (mbsherma) [reply process_log.txt:1]: Resolved in commit [dc6ae47](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/dc6ae47901cac08f0963bcbbfc0928c969a64c15)

File: process_server/process_log.txt
Comment (mbsherma) [reply process_server/process_log.txt:1]: Resolved in commit [dc6ae47](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/dc6ae47901cac08f0963bcbbfc0928c969a64c15)

File: scripts/process_log.txt
Comment (mbsherma) [reply scripts/process_log.txt:1]: Resolved in commit [dc6ae47](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/dc6ae47901cac08f0963bcbbfc0928c969a64c15)

File: scripts/kill_NSFRB
Comment (mbsherma) [reply scripts/kill_NSFRB:1]: Resolved in commit [dc6ae47](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/dc6ae47901cac08f0963bcbbfc0928c969a64c15)

File: scripts/run_classifier.py
Comment (mbsherma) [reply scripts/run_classifier.py:1]: Resolved in commit [dc6ae47](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/dc6ae47901cac08f0963bcbbfc0928c969a64c15)

File: scripts/run_NSFRB
Comment (mbsherma) [reply scripts/run_NSFRB:1]: Resolved in commit [dc6ae47](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/dc6ae47901cac08f0963bcbbfc0928c969a64c15)

File: process_server/run_COMBINED.py
Comment (mbsherma) [reply process_server/run_COMBINED.py]: Resolved in commit [22e0d71](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/22e0d71b5da4fb2366d38908bd35f2c16f8625e1)

File: tmpoutput/all_cluster_cands.csv
Comment (mbsherma) [reply tmpoutput/all_cluster_cands.csv:1]: I can work on this after merging

File: tmpoutput/search_log_thread_row2_col1.txt
Comment (mbsherma) [reply tmpoutput/search_log_thread_row2_col1.txt:1]: I can work on this after merging

**Comment by mbsherma** [comment]
> @caseyjlaw @nkosogor @cpviolator Overall there's no merge conflicts after my first review; however, I've found a few instances of dead code that I need to remove, and multiple log files I've made that are cluttering the repo. I've commented these above and assigned them to myself to resolve. @nkosogor I'm going to work on removing some dead code and clearing the log files and can approve once i've done that. Once its merged I'll take a closer look at how I'm maintaining the log files. There's also a small bug in the process server with how I'm using concurrent.futures that I'll fix before we merge

I've addressed these issues and committed them to the pull request. @nkosogor be sure to check that I didn't delete any log files or dead code that you needed, but I don't think I did. I'll do one more pass to review and can approve the pull request unless I notice anything else.

File: nsfrb/plotting.py
Comment (mbsherma) [inline nsfrb/plotting.py:1]: @mbsherma You should move searching.search_plots_new to nsfrb/plotting.py for consistency

File: nsfrb/searching.py
Comment (mbsherma) [inline nsfrb/searching.py]: @mbsherma Move this to nsfrb/plotting.py

File: scripts/README.md
Comment (mbsherma) [inline scripts/README.md]: @mbsherma remove this b/c testing moved to tests folder

File: scripts/README.md
Comment (mbsherma) [inline scripts/README.md]: @mbsherma remove this b/c testing moved to tests folder

File: nsfrb/searching.py
Comment (mbsherma) [reply nsfrb/searching.py]: Resolved in commit [78cb0ed](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/78cb0edfc85ffdf445089667f71141e579d956f9)

File: scripts/README.md
Comment (mbsherma) [reply scripts/README.md]: Resolved in commit [78cb0ed](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/78cb0edfc85ffdf445089667f71141e579d956f9)

File: scripts/README.md
Comment (mbsherma) [reply scripts/README.md]: Resolved in [78cb0ed](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/78cb0edfc85ffdf445089667f71141e579d956f9)

**Review by mbsherma — APPROVED** [review]
@nkosogor @caseyjlaw @cpviolator After implementing the changes discussed above (removing dead code, clearing log files, fixing multithreading in search, and moving plotting function to nsfrb/plotting), everything looks good to merge! I'll address the other comments (restructuring logging, etc.) after the merge.

**Comment by cpviolator** [comment]
Hi All, I can't comment much yet on the actual code but I like the way this PR was handled. It's good to keep PRs to about 20-30 commits or <5k lines of code, but that's never a hard and fast rule. It's also good to resolve issues with git commit tags so everyone can see the individual commit that resolved an individual issue, great to see that here.

My other piece of feedback is that it would be good to have some information about the scope of the PR that bullet point and describe the changes and features added. Commits are best kept grouped together with a unified objective so that review is as streamline as possible, so if one finds oneself describing too many disparate things its a good indicator to break the commits into multiple PRs.

Again, this is looking very professional and clean.

**Comment by nkosogor** [comment]
I have updated the PR with the necessary information regarding the scope, changes and features added, so proceeding to merge now.

**Pull request merged**
